src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -437,6 +437,12 @@ public static unsafe IPPacketInformation GetIPPacketInformation(Interop.Winsock.
 
         public static unsafe IPPacketInformation GetIPPacketInformation(Interop.Winsock.ControlDataIPv6* controlBuffer)
         {
+            if (controlBuffer->length == (UIntPtr)sizeof(Interop.Winsock.ControlData))
```

Comment:
Why is this necessary? It looks like the callers already check this and call the right version of GetIPPacketInformation.

Reply:
not really. We check for the socket type and we pre-allocate the buffer. But for dual mode we can get get either one when the receive operation is finished. Pure IPv4 or Ipv6 does not have this problem as the controls buffer will always match the protocol family.

Reply:
BTW this check the length returned inside of the structure, not size of structure it self (like we do in some other places)

Reply:
Ah, that's what confused me. Callers are checking the size of the structure itself, not the returned length.